### PR TITLE
fix: fix data codec writer close

### PR DESCRIPTION
**Pull request by shaoting-huang**
issue:#33813

### CODE REVIEW

File: internal/storage/data_codec.go [hunk 1]
```diff
@@ -1324,3 +1261,28 @@ func (dataDefinitionCodec *DataDefinitionCodec) Deserialize(blobs []*Blob) (ts [
...
 }
+
+func handleWriterError(eventWriter *insertEventWriter, writer *InsertBinlogWriter, err error) ([]*Blob, error) {
+	if eventWriter != nil {
+		eventWriter.Close()
+	}
+	if writer != nil {
+		writer.Close()
+	}
+	return nil, err
```
Comment (sunby) [inline internal/storage/data_codec.go:1272]: no need to always return nil here.

File: internal/storage/data_codec.go [hunk 2]
```diff
@@ -1324,3 +1261,28 @@ func (dataDefinitionCodec *DataDefinitionCodec) Deserialize(blobs []*Blob) (ts [
...
+	UniqueID,
+	error,
+) {
+	if eventReader != nil {
+		eventReader.Close()
+	}
+	if binlogReader != nil {
+		binlogReader.Close()
+	}
+	return InvalidUniqueID, InvalidUniqueID, InvalidUniqueID, err
```
Comment (sunby) [inline internal/storage/data_codec.go:1287]: same as above

File: internal/storage/data_codec.go (hunk 1 above)
Comment (shaoting-huang) [reply internal/storage/data_codec.go]: fix

**Comment by shaoting-huang** [comment]
rerun ut

**Comment by shaoting-huang** [comment]
rerun ut

**Pull request merged**